Assets/Scripts/Player/PhysicsHands/GrabLogic.cs

```diff
@@ -0,0 +1,144 @@
+            {
+                var dist = Vector3.Magnitude(_grabbedRb.position - _grabCenter.position) - _initialGrabDist;
+                if(dist > .5 * _grabRange)
+                {
+                    EndGrab();
+                }
+            }
+
+        }
+
+
+        public void CheckForGrabbables()
+        {
+            Debug.Log("i'm gonna grab u!!!!1");
+            if (_grabbedRb == null)
+            {
+                //do multi sphere cast from the grabbing center, then get the closest object. This becomes _grabbedRb
+                RaycastHit[] spherecasthits;
+                List<Rigidbody> potentialRigidbodies = new List<Rigidbody>();
+
+                
+                
+                spherecasthits = Physics.SphereCastAll(_grabCenter.position, _grabRange, Vector3.forward, _grabRange,_grabLayerMask);
+
+                foreach (RaycastHit hit in spherecasthits)
+                {
+                    GameObject hitObject = hit.collider.gameObject;
+                    Rigidbody oldrb;
+                    if (hitObject.TryGetComponent<Rigidbody>(out oldrb))
+                    {
+                        //oldrb.AddExplosionForce(800, _grabCenter.position, _grabRange);
+                        if (!oldrb.isKinematic) {
+
+                            potentialRigidbodies.Add(oldrb);
+                            
+                        }
+                    }
+                }
+
+                if (potentialRigidbodies.Count > 0)
+                {
+                    _grabbedRb = getClosestRigidbody(_grabCenter, potentialRigidbodies);
+                    _initialGrabDist = Vector3.Magnitude(_grabbedRb.position - _grabCenter.position);
+                    StartGrab();
+                    Debug.Log(_grabbedRb.gameObject.name);
+                }
+            }
+        }
+        private void StartGrab() 
+        { 
+            //attach a configurable joint to the grabbed rb and configure it
+            _grabbedJoint = _grabbedRb.gameObject.AddComponent<ConfigurableJoint>();
+            _grabbedJoint.connectedBody = _grabbingRb;
+            _grabbedJoint.xMotion = ConfigurableJointMotion.Locked;
+            _grabbedJoint.yMotion = ConfigurableJointMotion.Locked;
+            _grabbedJoint.zMotion = ConfigurableJointMotion.Locked;
+            _grabbedJoint.angularXMotion = ConfigurableJointMotion.Locked;
+            _grabbedJoint.angularYMotion = ConfigurableJointMotion.Locked;
+            _grabbedJoint.angularZMotion = ConfigurableJointMotion.Locked;
+            _oldGrabbableLayer = _grabbedJoint.gameObject.layer;
+            _grabbedJoint.gameObject.layer = 14;
+        }
+
+        public void EndGrab()
+        {
+            if (_grabbedRb != null) { 
+                Debug.Log("i'm gonna stop grab u!!!!1");
+                _grabbedJoint.gameObject.layer = _oldGrabbableLayer;
+                //remove a configurable joint
+                Destroy(_grabbedJoint);
```

Comment:
Would it be possible to reuse joints? If not ignore this comment

Reply:
Since the joints are put on the grabbed objects, no.